bitbots_bringup/scripts/check_robot.py

```diff
@@ -0,0 +1,285 @@
+    launch.start()
+    while True:
+        if is_motion_started():
+            rospy.sleep(5)
+            break
+        else:
+            print("Waiting for software to be started \n")
+            rospy.sleep(1)
+    print("\n\n")
+
+    # check diagnostic status
+    print("Will check diagnostic status of robot.\n")
+    diag_sub = rospy.Subscriber("/diagnostics_toplevel_state", DiagnosticStatus, diagnostic_cb, tcp_nodelay=True)
+    rospy.sleep(3)
+    if diag_status == DiagnosticStatus.OK:
+        print("    Diagnostic status okay.")
+    else:
+        print_warn("    Diagnostics report problem. Please use rqt_monitor to investigate.")
+    print("\n\n")
+
+    # check publishing frequency of imu, servos, pwm, goals, pressure, robot_state
+    # the topics which will be checked with minimal publishing rate
+    topics_to_check = {"/imu/data": 900, "/joint_states": 900, "/robot_state": 900, "/foot_pressure_left/raw": 900,
+                       "/foot_pressure_left/filtered": 900, "/foot_pressure_right/raw": 900,
+                       "/foot_pressure_right/filtered": 900, "/core/vdxl": 9, "/diagnostics_toplevel_state": 9}
+    rts = []
+    for topic in topics_to_check.keys():
+        msg_class, real_topic, _ = rostopic.get_topic_class(topic)
+        if real_topic is None or msg_class is None:
+            print_warn(F"Problem with topic {topic}")
+        else:
+            rt = rostopic.ROSTopicHz(-1)
+            rt_sub = rospy.Subscriber(topic, msg_class, rt.callback_hz, callback_args=topic, tcp_nodelay=True)
+            rts.append((rt, rt_sub))
+    print("Please wait a few seconds for publishing rates of topics to be evaluated\n")
+    rospy.sleep(5)
+    print("Topics have been evaluated:\n")
+    i = 0
+    for topic in topics_to_check.keys():
+        rate = rts[i][0].get_hz(topic)[0]
+        if rate is None or rate < topics_to_check[topic]:
+            print_warn(F"  Low rate on Topic {topic}: {round(rate, 2)} \n")
+        else:
+            print(F"  Okay rate Topic {topic}: {round(rate, 2)} \n")
+        i += 1
+    # todo check if some of the topics have very heterogeneous msg receive times
+
+    # check tf
+    # todo do some kind of check to see if tf tree is connected
+
+    # check pressure values when robot in air
+    print("\n\n")
+    print("We will check the foot pressure sensors next\n")
+    input("Please hold the robot in the air so that the feet don't touch the ground and press enter.")
+    left_pressure_sub = rospy.Subscriber("/foot_pressure_left/filtered", FootPressure, pressure_cb, callback_args=True,
+                                         tcp_nodelay=True)
+    right_pressure_sub = rospy.Subscriber("/foot_pressure_right/filtered", FootPressure, pressure_cb,
+                                          callback_args=False,
+                                          tcp_nodelay=True)
+    rospy.sleep(0.5)
+    while (not left_pressure) and (not right_pressure):
```

Comment:
Should this time out eventually, in case one sensor is not working at all?

Reply:
I think it should not timeout. If you don't get any messages you will have to fix it. Just added a check for shutdown to make ctrl+c possible.